nvidia support

dasha-uwu opened the pull request:
the dummy pointers are a big ugly but i see no other way of doing it

metux commented:
Thanks for the great puzzling work :)

Just needs a bit of pedantic editorial polishing, in order to document it all well for the future (see above comments) - then have a go.

dasha-uwu commented on dix/privates.c:
updated commit message

dasha-uwu commented on Xext/xvdix.h:
some of these cause segfaults in nvidia code, some are preemptive
i really wouldn't want to debug another nvidia segfault if some of the preemptive turn out to be neccessary...
if the alternative is just adding dummy ptr everywhere it affects ABI

dasha-uwu commented on os/client_priv.h:
```diff
@@ -53,7 +53,7 @@ void NotifyParentProcess(void);
 void ResetWellKnownSockets(void);
 void CloseWellKnownConnections(void);
-void SetCriticalOutputPending(void);
+_X_EXPORT void SetCriticalOutputPending(void);
```
sorry i don't understand what else can i say in the commit message? it's already saying it's for nvidia

metux replied:
See https://github.com/X11Libre/xserver/commit/41c039948f1464ea9675ebc96f84afe314eb1dcd

The reason why I'm so nitpicking here:
Next time we're revising/changing the module ABI, we might fall into the same trap again, because w/o the source code, we have no idea what they're doing :(
By adding some comment right into the header, it's immediately clear why those symbols need to be exported.

dasha-uwu replied:
> Can't you run readelf/objdump/nm on the proprietary driver to see what it's importing? Dump it to a text file and keep that around for reference.

this is precisely what i did. i've exported all symbols obtained this way.

dasha-uwu replied:
the nvidia driver also dlsym's some stuff so that can't be used to find everything.

dasha-uwu commented:
i've added comments and moved from priv to nonpriv headers where possible. some nonpriv headers were removed though, should i revert that and put the exported functions there?

catfromplan9 commented:
@metux bump
